### ISSUE TITLE
Backport Ruby 2.5 support to Rails 5.0

### ISSUE THREAD

**Issue by booch** [rails/rails#31478]
My team is hoping to upgrade to Ruby 2.5 before upgrading to Rails 5.1. It appears that there are some fixes on the 5.1 and master branches that haven't been made in Rails 5.0. I'm hoping it's useful to track those here, and that they can be cherry-picked into the 5.0 branch.

### System configuration
**Rails version**: 5.0.6

**Ruby version**: 2.5

**Comment by booch** [comment]
* https://github.com/rails/rails/commit/3d91649654ff5570d0cf26f96ec9f82f31fe32f6
* https://github.com/rails/rails/pull/31099/commits/4022f33416a00fd6a18989aa54f60573f48be0a0
* https://github.com/rails/rails/pull/31099/commits/2d5700b9a3e2ac25a5ff349ca6d3dc27077cea13
* https://github.com/rails/rails/pull/31455/commits/94c1bacd9b515bb0fd9aa231140a8e1f45a26574
* https://github.com/rails/rails/pull/31462/commits/59a9951d202da4ce2cbd042245d0294d6c3d4329

**Comment by rafaelfranca** [comment]
Rails 5.0 is in its EOL. We are going to backport some change to support 2.5 and release a last version but we are not going to push any other official release, even if it is a security issue.

**Comment by dla-c-box** [comment]
Hi @rafaelfranca .  When you said "We are going to backport some change **to support 2.5** and release a last version", that last version was 5.0.7, correct?  And you did what you said you would do and added the backports to support 2.5, correct?
I'm wondering, because [this](https://www.fastruby.io/blog/ruby/rails/versions/compatibility-table.html) Rails/Ruby compatibility matrix mentions that Rails 5.0.Z requires Ruby >= 2.2.2, **< 2.5.0** , (refering to the current issue #31478 to back this claim), but based on your comment and the fact that the breaking changes proposed here seem to have been backported e.g. [here](https://github.com/rails/rails/blob/v5.0.7.2/activesupport/test/core_ext/module_test.rb#L30) and [here](https://github.com/rails/rails/blob/v5.0.7/actionpack/test/abstract_unit.rb#L445), it seems it's on purpose that e.g. you didn't change the ruby dependency to be `< 2.5.0` on Rails 5.0.7.2 when it was released on 2019-03-13?

i.e. can we trust the fact that the official Rails 5.0.7.2 doc said it required Ruby `>= 2.2.2` and said nothing about Ruby `< 2.5.0` (although it had been around for 2 years at the time of that last release), or was it just an oversight at that time to not fix the Ruby dependency to become `< 2.5.0` ?

**Comment by rafaelfranca** [comment]
Rails doesn't define upper bounds for any dependency, even Ruby. That allow you to try to use new Ruby versions in your application. That means that you will never see in any official Rails documentation things like:

> You can't use this version of Rails with this version of Ruby that was released several years after the Rails version was released.

I indeed backported the the changes requested, but I didn't verified it indeed work with Ruby 2.5. Like I said, Rails 5.0 was not supported at the time anymore, so we have no obligation to keep developing it to support new Ruby version.

To be safe, use the version of Ruby that exited when that version of Rails was released.
